src/renderer/utils/storageHelper.ts

```diff
@@ -21,46 +21,61 @@ export interface StorageHelperOptions<T> {
   autoInit?: boolean; // start preloading data immediately, default: true
   observable?: CreateObservableOptions;
   storage: StorageAdapter<T>;
-  defaultValue?: T;
+  defaultValue: T;
 }
 
 export class StorageHelper<T> {
-  static defaultOptions: Partial<StorageHelperOptions<any>> = {
+  static readonly defaultOptions: Partial<StorageHelperOptions<any>> = {
     autoInit: true,
     observable: {
       deep: true,
       equals: comparer.shallow,
     }
   };
 
-  @observable private data = observable.box<T>();
+  private data: IObservableValue<T>;
   @observable initialized = false;
   whenReady = when(() => this.initialized);
 
-  get storage(): StorageAdapter<T> {
-    return this.options.storage;
-  }
-
-  get defaultValue(): T {
-    return this.options.defaultValue;
-  }
+  public readonly storage: StorageAdapter<T>;
+  public readonly defaultValue: T;
 
   constructor(readonly key: string, private options: StorageHelperOptions<T>) {
-    this.options = { ...StorageHelper.defaultOptions, ...options };
-    this.configureObservable();
-    this.reset();
+    this.data = observable.box<T>(this.options.defaultValue, {
+      ...StorageHelper.defaultOptions.observable,
+      ...(options.observable ?? {})
+    });
+    this.data.observe(change => {
+      const { newValue, oldValue } = toJS(change, { recurseEverything: true });
+
+      this.onChange(newValue, oldValue);
+    });
+
+    this.storage = options.storage;
+    this.defaultValue = options.defaultValue;
 
     if (this.options.autoInit) {
       this.init();
     }
   }
 
+  /**
+   * This function turns an optionally synchronous call to an asynchronous call
+   * and correctly handles the synchronous call throwing an error
+   * @returns the result of this.storage.getItem
+   */
+  private async getItemFromStorage(): Promise<T> {
```

Comment:
Btw, I guess this is not correct at all and breaks work with `sync` reading from storage.

See `storageHerlper.test.ts`:

```ts
    it("initialized with default value", async () => {
      localStorage.setItem(storageKey, "saved"); // pretending it was saved previously

      expect(storageHelper.key).toBe(storageKey);
      expect(storageHelper.defaultValue).toBe("test");
      expect(storageHelper.get()).toBe("test");

      // here actually using of await is wrong and if you remove it test will fail which is not happening in previous version
      // for that reason was introduced cumbersome `loadFromStorage()` which is handle it correctly.
      await storageHelper.init();
```